arrows/ocv/inpaint.cxx

```diff
@@ -0,0 +1,169 @@
+{
+  attach_logger( "arrows.ocv.inpaint" );
+}
+
+// ----------------------------------------------------------------------------
+inpaint::~inpaint()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+inpaint
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "inpaint_method", method_converter().to_string(
+                       d->method ),
+                     "Inpainting method, possible values: " +
+                     method_converter().element_name_string() );
+  config->set_value( "radius", d->radius,
+                     "Radius parameter for the inpainting method" );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+inpaint
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Starting with our generated vital::config_block to ensure that assumed
+  // values are present. An alternative is to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for filtering
+  d->method = config->get_enum_value< method_converter >( "inpaint_method",
+                                                          d->method );
+  d->radius = config->get_value< float >( "radius", d->radius );
+}
+
+// ----------------------------------------------------------------------------
+bool
+inpaint
+::check_configuration( vital::config_block_sptr config ) const
+{
+  auto const radius = config->get_value< float >( "radius" );
+  if( radius <= 0 )
+  {
+    LOG_ERROR( logger(),
+               "Radius should be positive but instead was " << radius );
+    return false;
+  }
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+kwiver::vital::image_container_sptr
+inpaint
+::filter( kwiver::vital::image_container_sptr image )
+{
+  cv::Mat cv_image = ocv::image_container::vital_to_ocv(
+    image->get_image(), ocv::image_container::RGB_COLOR );
+
+  if( cv_image.channels() != 4 )
+  {
+    LOG_ERROR(
+      logger(),
+      "Expected 4 image channels but instead there were " <<
+        cv_image.channels() );
+    return image;
+  }
+  cv::Mat rgba_channels[ 4 ];
+  cv::split( cv_image, rgba_channels );
+  cv::merge( rgba_channels, 3, cv_image ); // merge the first three channels
+  switch( d->method )
+  {
+    case METHOD_telea:
+    {
+      cv::inpaint( cv_image, rgba_channels[ 3 ], cv_image, d->radius,
+                   cv::INPAINT_TELEA );
+      break;
+    }
+    case METHOD_navier_stokes:
+    {
+      cv::inpaint( cv_image, rgba_channels[ 3 ], cv_image, d->radius,
+                   cv::INPAINT_NS );
+      break;
+    }
+    case METHOD_mask:
+    {
+      cv::Mat zeros = cv::Mat::zeros( cv_image.size(), cv_image.type() );
```

Comment:
I guess it's not necessary to do now, but maybe as a follow-up, should we allow the user to configure the inpainting color?

Reply:
I agree this could be useful later. If you have an example handy of specifying a multi-element config value, since I'd probably let the user specify an RGB tuple, that would be useful.

Reply:
I'm sure we could find a cleaner way, but here is an example:
https://github.com/Kitware/kwiver/blob/master/arrows/cuda/integrate_depth_maps.cxx#L160

We just read/write the value as a string and then convert it with iostream